onnxscript/ir/passes/common/_c_api_utils.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Utilities for interfacing with onnx C APIs."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Callable
+
+from onnxscript import ir
+
+if TYPE_CHECKING:
+    import onnx
+
+
+logger = logging.getLogger(__name__)
+# Temporarily remove initializers larger than this size to keep model size down
+# for the onnx.shape_inference call because it needs to serialize the model
+_BIG_TENSOR_SIZE_LIMIT = 1000  # 1KB
+
+
+def call_onnx_api(
+    func: Callable[[onnx.ModelProto], onnx.ModelProto],
+    model: ir.Model,
+    merge_func: Callable[[ir.Model, onnx.ModelProto], tuple[ir.Model, bool]],
+) -> tuple[ir.Model, bool]:
+    """Call an ONNX C API function by temporarily removing initializers.
+    This is necessary because the ONNX C API does not support large models
+    with initializers that have large tensor values.
+
+    Args:
+        func: Partially applied function that takes a model proto and returns a model proto.
+        model: The IR model to pass to the API function.
+        merge_func: Function that merges IR model with information from the model proto.
+
+    Returns:
+        A tuple containing the modified model and a boolean indicating whether the model was modified.
+    """
+
+    # Store the original initializer values so they can be restored
+    initializer_values = tuple(model.graph.initializers.values())
+    tensors = {v.name: v.const_value for v in initializer_values}
+    original_inputs_len = len(model.graph.inputs)
+    initializer_names = {v.name for v in initializer_values}
```

Comment:
nit: Looks like the check is line 73 is not really being used effectively? Wouldn't it be better to ensure that we create only a list of the temporarily removed initializers in the loop below?

Reply:
I updated the function to use a dictionary and removed the check. Does it look like what you had in mind?